plugins/modules/system/java_keystore.py

```diff
@@ -379,7 +511,21 @@ def main():
         add_file_common_args=spec.add_file_common_args,
     )
     module.run_command_environ_update = dict(LANG='C', LC_ALL='C', LC_MESSAGES='C')
-    process_jks(module)
+
+    jks = JavaKeystore(module)
+
+    if jks.exists():
+        if module.params['force']:
+            jks.create()
+        else:
+            if jks.cert_changed():
+                jks.create()
+            else:
+                if not module.check_mode:
+                    jks.update_permissions()
```

Comment:
This should also be run in non-check_mode and return whether something was changed or not.

Reply:
I had not checked for existing bugs, but with your suggestions `changed` now reflects the correct status with both `check_mode` enabled and disabled for changes to the common file parameters.